src/app/features/city-staff-permissions/change-account-permission/change-account-permission.component.spec.ts

```diff
@@ -38,24 +42,177 @@ describe('ChangeAccountPermissionComponent', () => {
+  });
+  beforeEach(async () => {
     fixture = TestBed.createComponent(ChangeAccountPermissionComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+    _changeAccountPermissionService = fixture.debugElement.injector.get(
+      ChangeAccountPermissionService
+    );
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should close the dialog if the "success" element emits', () => {
+    const functionSpy = spyOn(dialogMock, 'close');
+    spyOnProperty(
+      _changeAccountPermissionService,
+      'success$',
+      'get'
+    ).and.returnValue(
+      of({
+        __typename: 'User',
+        email: 'example@test.com',
+        firstName: 'exampleText',
+        lastName: 'exampleTest',
+        permissions: AccessLevel.ADMIN,
+        username: 'exampleText',
+      })
+    );
+    fixture.detectChanges();
+    expect(functionSpy).toHaveBeenCalled();
+  });
+
+  it('should call the funcion "updateUserAccessInput" in the service when submit function is called in the component and the form is valid', () => {
+    const functionSpy = spyOn(
+      _changeAccountPermissionService,
+      'updateUserAccessInput'
+    );
+
+    component.formGroup.setValue({
+      type: 'GUEST',
+    });
+    fixture.detectChanges();
+    component.submit();
+    expect(functionSpy).toHaveBeenCalledOnceWith({
+      permissions: AccessLevel.GUEST,
+      username: 'id',
+    });
+  });
+
+  it('should show a error in the form when submit function is called in the component and the form is invalid', () => {
+    component.formGroup.setValue({
+      type: '',
+    });
+
+    component.submit();
+    fixture.detectChanges();
+    expect(
+      fixture.debugElement.nativeElement.querySelectorAll('mat-error').length
+    ).toEqual(1);
+  });
+
+  it('should show the loading bar when the data is loading', () => {
+    spyOnProperty(
+      _changeAccountPermissionService,
+      'loading$',
+      'get'
+    ).and.returnValue(of(true));
+    fixture.detectChanges();
+
+    const dpLoadingBars =
+      fixture.debugElement.nativeElement.querySelectorAll('dp-loading-bar');
+
+    expect(dpLoadingBars.length).toBe(1);
+  });
+
+  it('should not show the loading bar if the data is not loading', () => {
+    spyOnProperty(
+      _changeAccountPermissionService,
+      'loading$',
+      'get'
+    ).and.returnValue(of(false));
+
+    fixture.detectChanges();
+
+    const dpLoadingBars =
+      fixture.debugElement.nativeElement.querySelectorAll('dp-loading-bar');
+
+    expect(dpLoadingBars.length).toBe(0);
+  });
+
+  it('should show the error element if an error ocurred and is not loading', () => {
+    spyOnProperty(
+      _changeAccountPermissionService,
+      'error$',
+      'get'
+    ).and.returnValue(of('error'));
+
+    spyOnProperty(
+      _changeAccountPermissionService,
+      'loading$',
+      'get'
+    ).and.returnValue(of(false));
+
+    component.ngOnInit();
```

Comment:
Please remove this ngOnInit call. It is not needed before calling detectChanges.